README.md

```diff
@@ -1,2 +1,214 @@
+
+* `net_handler` communicates with both Dawn and Shepherd. It receives start and stop commands, input data for the coding challenge, and information about the gamepad state. It sends log messages, data about the connected devices, and output data for the coding challenge. This data is sent over both TCP and UDP connections (depending on the type of data) to Shepherd and Dawn, and the data is packaged using Google Protocol Buffers.
+* `dev_handler` and `net_handler` are connected via `shm_wrapper`. This connection is used for the `net_handler` to fetch the most recent state of the connected devices, which is then sent to Dawn for the students to see.
+* `executor` and `net_handler` are connected via `shm_wrapper_aux` and a FIFO pipe. The connections are used to pass information about the coding challenge (both inputs to the functions and the student's outputs), and for `executor` to know what to run at any given time (autonomous mode, teleop mode, coding challenges, or idle).
+* `executor` and `dev_handler` are connected via `shm_wrapper`. This connection is used for `executor` to send commands to the attached devices (tell a motor to run at a certain speed, move a servo to certain position, etc.), and for `dev_handler` to serve `executor` with the device data that it needs to run the student code.
+* `dev_handler` communicates with `lowcar` via serial connection. This connection is used to poll for new devices, detect when devices have disconnected, and send device data and commands between the Raspberry Pi and the Arduinos.
+* All three processes output log messages through the `logger` tool. If logs are to be sent over the network to Dawn, those logs are put into a FIFO pipe which is opened by `net_handler`, where the logs are processed and sent to Dawn.
+
+## Concepts
+
+Runtime uses a **lot** of systems concepts, which is hopefully why you're here--to learn about systems! The following is a list of some various concepts that you'll be exposed to in Runtime, along with where they are used within Runtime. Keep in mind, this is _not_ a list of prerequisite knowledge you need to have in order to work on Runtime!
+
+* **Networking**
+    * Used to connect Runtime to Dawn and Shepherd
+    * **Unreliable Datagram Protocol (UDP)**
+	    * This networking standard for transmitting data between two machines on a network is useful when transferring packets of data where dropping, reordering, or duplicating packets occasionally is tolerable
+		* In Runtime, used for Runtime to communicate device data to Dawn, and Dawn to communicate gamepad state to Runtime
+	* **Transmission Control Protocol (TCP)**
+		* This networking standard for transmitting data between two machines on a network is useful when transferring packets of data that needs to be reliably sent and received
+		* In Runtime, used for Runtime to communicate all other data between itself and Dawn or Shepherd
+	* **Sockets**
```

Comment:
I actually just changed the challenge data FIFO pipe into a UNIX socket since its better/easier for two-way reply/respond communication. Probably need to adjust those 2 items here.

Reply:
Ok, I'll adjust it here, but keep in mind that this README is definitely gonna be a living doc, and most likely more than a few things in this doc are gonna have to be changed by the time we have a production version!